### PR TITLE
Added alias support for absolute imports

### LINKED ISSUE #253: Refactor frontend to use absolute imports
**Is your feature request related to a problem? Please describe.**

> I've been reviewing some frontend PRs recently and noticed frequent use of `../../../` in TypeScript imports. Is there a more efficient or cleaner way to handle these relative imports, especially for deeply nested files? Hopefully, for both tests and the source code:

image

**Describe the solution you'd like**
Refactor both frontend code and its tests to use absolute imports w/o `../../../`.

### PR DESCRIPTION
**Pull request by yashpandey06**
Resolves #253 

Added TypeScript path aliases to streamline module imports. Updated `tsconfig.json` and `vite.config.ts` for easier imports across the project.

### CODE REVIEW

**Comment by yashpandey06** [comment]
@arkid15r Please help review these changes .

Thanks !

**Comment by yashpandey06** [comment]
@arkid15r  Please help review these changes  !

File: frontend/__tests__/src/pages/Chapters.test.tsx [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
 import React from 'react'
 
-import { fetchAlgoliaData } from '../../../src/lib/api'
-import { render } from '../../../src/lib/test-util'
+import { fetchAlgoliaData } from '@src/lib/api'
+import { render } from '@src/lib/test-util'
 import '@testing-library/jest-dom'
-import { ChaptersPage } from '../../../src/pages'
+import { ChaptersPage } from '@src/pages'
 import { mockChapterData } from '../data/mockChapterData'
```
Comment (arkid15r) [inline frontend/__tests__/src/pages/Chapters.test.tsx]: And this case looks like a good candidate for `@tests` alias -- @tests/data/...

Comment (yashpandey06) [reply frontend/__tests__/src/pages/Chapters.test.tsx]: Sure ...let's make it like that ! 

Thanks for the review !

File: frontend/tsconfig.json [hunk 2]
```diff
@@ -7,16 +7,25 @@
...
+    "inlineSources": true,
+    "baseUrl": ".",
+    "paths": {
+      "@src/*": ["src/*"],
+      "components/*": ["src/components/*"],
+      "utils/*": ["src/utils/*"],
+      "lib/*": ["src/lib/*"],
+      "pages/*": ["src/pages/*"],
+      "@test/*": ["__tests__/src/*"]
+    }
```
Comment (arkid15r) [inline frontend/tsconfig.json]: Is it possible to have baseUrl set to `src` and in paths keep `@tests` alias only?

Comment (yashpandey06) [reply frontend/tsconfig.json:26]: Hey ! ..i tried doing so ...so when base url is src the test folder which is out side the src folder  is not recognising the paths that we are  mentioning .

Comment (yashpandey06) [reply frontend/tsconfig.json]: Yeah in the latest commits I tried to do it ...but jest tests were failing for such configuration they weren't able to point to src .. so when I explicitly mentioned the src in ts config along with @test and also added to the jest config ...
Then everything was running fine .

Comment (yashpandey06) [reply frontend/tsconfig.json]: Tbh ...it was difficult to debug the actual findings of the errors .

Comment (arkid15r) [reply frontend/tsconfig.json]: Yes, keeping code base nice and tidy isn't easy. It requires some extra time and skills (also stubbornness sometimes) to get it done.

I've updated the PR with how I wanted imports to be reconfigured -- please review when you get a chance.
Thank you!

Comment (yashpandey06) [reply frontend/tsconfig.json]: @arkid15r LGTM

Comment (yashpandey06) [reply frontend/tsconfig.json]: we have switched to baseUrl --. pattern again !

Comment (arkid15r) [reply frontend/tsconfig.json]: > @arkid15r LGTM

This code doesn't work in the app even though the tests are fine

Comment (yashpandey06) [reply frontend/tsconfig.json]: @arkid15r Yeah ...your right !...let me check

**Comment by yashpandey06** [comment]
@arkid15r ...the thing is vite needs explicit mention of paths while building

**Review by arkid15r — APPROVED** [review]

**Pull request merged**